forge-gui/res/cardsfolder/upcoming/cosmic_epiphany.txt

```diff
@@ -0,0 +1,7 @@
+Name:Cosmic Epiphany
+ManaCost:4 U U
+Types:Sorcery
+A:SP$ Draw | NumCards$ X | SpellDescription$ Draw cards equal to the number of instant and sorcery cards in your graveyard.
+SVar:X:Count$ValidGraveyard Instant.YouOwn,Sorcery.YouOwn
+DeckNeeds:Type$Instant|Sorcery & Ability$Graveyard
```

Comment:
Remove `Ability$Graveyard` from Needs – Instants and Sorceries get there on their own
Maybe `DeckHas:Ability$Graveyard` instead?